Simulation/Classes/Vehicle.m

```diff
@@ -0,0 +1,102 @@
+classdef Vehicle
+  properties(Access = protected)
+    stateVector = [];
+    controlsVector = [];
+    modelArray = {};
+  endproperties
+
+  properties (Constant)
+    SV_Index = StateVector_Index;
+	endproperties
+
+  properties(Access = public)
+    Name = '';
+  endproperties
+
+  methods
+    % Create Vehicle Object
+    %   Inputs:
+    %     Name = Name of Vehicle
+    %     init_stateVector = Initial State of Vehicle
+    %     N_Ctrls = Number of Control Inputs
+    %     modelArray = Array of Models that affect the Vehicle
+    function Vh = Vehicle(Name,init_stateVector,N_Ctrls,modelArray)
+      if (nargin != 4)

```

Comment:
Why is this necessary? Doesn't Matlab itself make sure that the correct number of arguments is passed in?

Reply:
Yes it does, however I wanted this to be more explicit and tell the user what the error is via a custom error message. I found this to be helpful when writing tools at work and thought this may help future users understand the code better.

Let me know your thoughts, I can remove it if you'd like, definitely not necessary.

Reply:
I have removed the error check since it is redundant. Kept the mass > 0 check since that is not something that Octave checks on its own.